lib/from_openstudio/program_type.rb

```diff
@@ -0,0 +1,75 @@
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE
+# UNITED STATES GOVERNMENT, OR THE UNITED STATES DEPARTMENT OF ENERGY, NOR ANY OF
+# THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+
+require 'honeybee/program_type'
+require 'to_openstudio/model_object'
+
+module Honeybee
+    class ProgramTypeAbridged < ModelObject
+
+        def self.from_programtype(programtype)
+            # create an empty hash
+            hash = {}
+            hash[:type] = 'ProgramTypeAbridged'
+            # set hash values from OpenStudio Object
+            hash[:identifier] = clean_name(programtype.nameString)
+            unless programtype.displayName.empty?
+                hash[:display_name] = (programtype.displayName.get).force_encoding("UTF-8")
+            end
+            #TODO: OS can have multiple loads assigned but honeybee schema has a single load in
+            #programtype
+            unless programtype.people.empty?
+                hash[:people] = Honeybee::PeopleAbridged.from_load(programtype.people[0])
+            end
+            unless programtype.lights.empty?
+                hash[:lighting] = Honeybee::LightingAbridged.from_load(programtype.lights[0])
+            end
+            unless programtype.electricEquipment.empty?
+                hash[:electric_equipment] = Honeybee::ElectricEquipmentAbridged.from_load(programtype.electricEquipment[0])
+            end
+            unless programtype.gasEquipment.empty?
+                hash[:gas_equipment] = Honeybee::GasEquipmentAbridged.from_load(programtype.gasEquipment[0])
+            end
+            #unless programtype.hotWaterEquipment.empty?
+            #    hash[:service_hot_water] = Honeybee::ServiceHotWaterAbridged.from_load(programtype.hotWaterEquipment[0])
+            #end
+            unless programtype.spaceInfiltrationDesignFlowRates.empty?
+                hash[:infiltration] = Honeybee::InfiltrationAbridged.from_load(programtype.spaceInfiltrationDesignFlowRates[0])
+            end
+            unless programtype.designSpecificationOutdoorAir.empty?
+                hash[:ventilation] = Honeybee::VentilationAbridged.from_load(programtype.designSpecificationOutdoorAir.get)
+            end
+            # TODO: add setpoint thermostat and humidistat? Not sure if a reverse translator would have a programtype setpoint object - These are assigned at the thermal zone level.
```

Comment:
@chriswmackey Do you have thoughts on this? These objects are only defined at the thermal zone level in the OS SDK and we are translating those to Honeybee rooms. How do we want to handle adding these objects to the programtype?

Reply:
I think it's fine if Ventilation and Setpoint objects are not assign-able to ProgramTypes when importing from OSM/IDF. Let's just assign it to the Rooms and that will ensure that the resulting simulations are the same between the imported IDF and the existing IDF.